formats/Gantt/src/Gantt.php

```diff
@@ -0,0 +1,311 @@
+		}
+	}
+
+	/**
+	 * Sort Tasks based on the sortkey
+	 *
+	 * @param GanttTask $a
+	 * @param GanttTask $b
+	 *
+	 * @return integer
+	 */
+	public function sortTasks( $a, $b ) {
+
+		$sortKey = $this->getSortKey();
+
+		// sort based on start/end date or title
+		if ( $sortKey == 'title' ) {
+			// sort based on title
+			return strcmp( $a->getTitle(), $b->getTitle() );
+		} else {
+			if ( strpos( $sortKey, 'date' ) !== false ) {
+				if ( strpos( $sortKey, 'start' ) !== false ) {
+					// sort based on startDate
+					return strcmp( $a->getStartDate(), $b->getStartDate() );
+				} else {
+					// sort based on endDate
+					return strcmp( $b->getEndDate(), $a->getEndDate() );
+				}
+			}
+		}
+	}
+
+
+	/**
+	 * Sort Sections based on the sortkey
+	 *
+	 * @param GanttSection $a
+	 * @param GanttSection $b
+	 *
+	 */
+	public function sortSections( $a, $b ) {
+
+		$sortKey = $this->getSortKey();
+
+		// sort based on start/end date or title
+		if ( $sortKey == 'title' ) {
+			// sort based on title
+			return strcmp( $a->getTitle(), $b->getTitle() );
+		} else {
+			if ( strpos( $sortKey, 'date' ) !== false ) {
+				if ( strpos( $sortKey, 'start' ) !== false ) {
+					// sort based on startDate
+					return strcmp( $a->getEarliestStartDate(), $b->getEarliestStartDate() );
+				} else {
+					// sort based on endDate
+					return strcmp( $b->getLatestEndDate(), $a->getLatestEndDate() );
+				}
+			}
+		}
+	}
+
+	private function createNewSection( $sectionID, $sectionTitle, $startDate, $endDate, $taskID ) {
+		$ganttSection = new GanttSection();
+		//check if the id in the object is realy needed or is it enough to have it as array key
+		$ganttSection->setID( $sectionID );
+		$ganttSection->setTitle( $sectionTitle );
+		$ganttSection->setEarliestStartDate( $startDate );
+		$ganttSection->setLatestEndDate( $endDate );
+		$ganttSection->addTask( $taskID );
+
+		$this->mSections[$sectionID] = $ganttSection;
+	}
+
+	/**
+	 * Creates output for mermaidjs
+	 *
+	 * @return string
+	 */
+	public function getGanttOutput() {
+
+		$sections = $this->getSections();
+		$tasks = $this->getTasks();
+		$sortKey = $this->getSortKey();
+
+		if ( !empty( $sortKey ) ) {
+			// Order Sections
+			usort( $sections, [ $this, "sortSections" ] );
+			usort( $tasks, [ $this, "sortTasks" ] );
+
+			// reorder TaskArray of current section
+			foreach ( $sections as $section ) {
+				$orderedTasks = [];
+				foreach ( $tasks as $task ) {
+
+					// check if $section->getTasks() holds ID of current task
+					if ( in_array( $task->getID(), $section->getTasks() ) ) {
+						$sectionTasks = $section->getTasks();
+						//loop through tasks of current section
+						foreach ( $sectionTasks as $taskKey => $sectionTask ) {
+							if ( $task->getID() == $sectionTask ) {
+								array_push( $orderedTasks, $sectionTask );
+							}
+						}
+					}
+				}
+				$section->setTasks( $orderedTasks );
+			}
+		}
+
+		/*
+		 * Bring the "section" with no title to the first position.

```

Comment:
Remove sort functionality, this can be done with inline query standard sort and order property
Line: 126-182

Commit: 6aa6b90db221170166efcb53f983a06dfe394b03